etc/kayobe/apt.yml

```diff
@@ -54,30 +54,46 @@ stackhpc_apt_repositories:
     suites: "{{ ansible_facts.distribution_release }} {{ ansible_facts.distribution_release }}-updates {{ ansible_facts.distribution_release }}-backports"
     components: main restricted universe multiverse
     architecture: amd64
-    required: true
+    required: "{{ os_release == 'jammy' }}"
   - url: "{{ stackhpc_repo_ubuntu_jammy_security_url }}"
     suites: "{{ ansible_facts.distribution_release }}-security"
     components: main restricted universe multiverse
     architecture: amd64
-    required: true
+    required: "{{ os_release == 'jammy' }}"
   - url: "{{ stackhpc_repo_ubuntu_jammy_cve_2024_6387_url }}"
     suites: "pulp"
     components: upload
     architecture: amd64
     trusted: yes
-    required: true
+    required: "{{ os_release == 'jammy' }}"
   - url: "{{ stackhpc_repo_docker_ce_ubuntu_jammy_url }}"
     suites: "{{ ansible_facts.distribution_release  }}"
     components: stable
     signed_by: docker.asc
     architecture: amd64
-    required: true
+    required: "{{ os_release == 'jammy' }}"
   - url: "{{ stackhpc_repo_ceph_reef_debian_url }}"
     suites: "{{ ansible_facts.distribution_release  }}"
     components: main
     signed_by: ceph.asc
     architecture: amd64
-    required: true
+    required: "{{ os_release == 'jammy' }}"
+  - url: "{{ stackhpc_repo_ubuntu_noble_url }}"
```

Comment:
nit: This section could be a bit neater if the jammy/noble repos were combined and the url was just selected based on the os_release. See [yoga](https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/yoga/etc/kayobe/apt.yml#L50) for how we did this before